org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/JavaSearchScope.java

```diff
@@ -374,6 +374,7 @@ private int indexOf(String fullPath) {
 		String currentRelativePath = this.relativePaths[i];
 		if (currentRelativePath == null) continue;
 		String currentContainerPath = this.containerPaths[i];
+		currentContainerPath = convertInternalToExternalPath(currentContainerPath, fullPath);
```

Comment:
I would avoid doing conversion on every container path in the loop during the `indexOf`, instead I would convert to the external paths immediately at creating the search containers and only convert once given path.
Same below.